code/modules/projectiles/pins.dm

```diff
@@ -187,3 +189,9 @@
 	if(gun)
 		gun.pin = null
 	return ..()
+
+obj/item/weapon/gun/Destroy()
```

Comment:
1. There is probably already a definition for destroy somewhere. You shouldn't put this with the pins.
2. You're not returning or even _calling_ parent.